test/functional-tests/pageObjects/searchPage.js

```diff
@@ -0,0 +1,140 @@
+Button = require('./webdriver-components/button.js')
+Dropdown = require('./webdriver-components/dropdown.js')
+BasePage = require('./basePage.js');
+NavBar = require('./ccd-components/globalNavBar.js');
+DateField = require('../pageObjects/ccd-components/fields/ccdDateField.js');
+let FieldUtils = require('../utils/fieldUtils.js');
+CaseList = require('./ccd-components/caseListComponent.js');
+
+class SearchPage extends BasePage {
+
+  constructor(){
+      super();
+      this._jurisdiction = new Dropdown('#s-jurisdiction');
+      this._caseType = new Dropdown('#s-case-type');
+      this._applyButton = new Button('.button', 'Apply');
+      this._resetButton = new Button('#reset');
+  }
+
+   /**
+    * Return a new instance of the Navigation bar component which is common to many
+    * pages and abstracted into it's own class
+    * @returns {NavBar|*}
+    */
+  getNavBarComponent(){
+    return new NavBar;
+  }
+
+  /**
+   * Return a new instance of the Case List Component which is common across both
+   * the case list page and search page
+   * @returns {CaseListComponent|*}
+   */
+  getCaseListComponent(){
+    return new CaseList;
+  } 
+
+  /**
+   * Select a Jurisdiction from the dropdown
+   * @param option to select - case insensitive
+   * @returns {Promise<void>}
+   */
+  async selectJurisdiction(option){
+    await this._jurisdiction.selectFromDropdownByText(option);
+  }
+
+  /**
+   * Select Case Type from the dropdown
+   * @param option to select - case insensitive
+   * @returns {Promise<void>}
+   */
+  async selectCaseType(option){
+    await this._caseType.selectFromDropdownByText(option);
+  }
+
+  /**
+   * Click Apply button to submit options for search
+   * @returns {Promise<void|*>}
+   */
+  async clickApplyButton() {
+    await this._applyButton.waitForElementToBeClickable();
+    await this._applyButton.click();
+    await browser.waitForAngular;
```

Comment:
does the test run and pass without this line: browser.waitForAngular? If so lets remove it

Reply:
Yes will do.